client/clip_client/client.py

```diff
@@ -361,20 +360,19 @@ def _iter_rank_docs(
                 )
 
     def _get_rank_payload(self, content, kwargs):
-        return dict(
+        payload = dict(
             on='/rank',
             inputs=self._iter_rank_docs(
                 content, _source=kwargs.get('source', 'matches')
             ),
             request_size=kwargs.get('batch_size', 8),
             total_docs=len(content) if hasattr(content, '__len__') else None,
-            metadata=_grpc_credential_wrapper(self.credential)
-            if self._scheme == 'grpc'
-            else None,
-            headers=_http_credential_wrapper(self.credential)
-            if self._scheme == 'http'
-            else None,
         )
+        if self._scheme == 'grpc':
+            payload.update(metadata=get_authentication(self._scheme, self.credential))
```

Comment:
how to process with the case of  `get_authentication()` returns None?

Reply:
None value wont affect